### PR TITLE
[Persistence] Update KVStore to use Badger wrapper functions

### PR DESCRIPTION
**Pull request by h5law**
## Description

This PR introduces unit tests to cover the KVStore's functionality as well as updating the KVStore logic to use the Badger wrapper functions `update`, `view`, etc.

This fixes the issues around deleting keys from the KVStore.

## Issue

Fixes N/A

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [x] Bug fix
- [x] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Update KVStore to use badger's wrapper functions
- Add KVStore unit tests

## Testing

- [x] `make develop_test`; if any code changes were made
- [x] `make test_e2e` on [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any code changes were made
- [x] `e2e-devnet-test` passes tests on [DevNet](https://pocketnetwork.notion.site/How-to-DevNet-ff1598f27efe44c09f34e2aa0051f0dd); if any code was changed
- [x] [Docker Compose LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md); if any major functionality was changed or introduced
- [x] [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any infrastructure or configuration changes were made

## Required Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added, or updated, [`godoc` format comments](https://go.dev/blog/godoc) on touched members (see: [tip.golang.org/doc/comment](https://tip.golang.org/doc/comment))
- [x] I have tested my changes using the available tooling
- [ ] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: persistence/kvstore/kvstore_test.go
Comment (Olshansk) [inline persistence/kvstore/kvstore_test.go:59]:
```suggestion
			name:     "Successfully retrieve a value from the store",
```

**Review by Olshansk — APPROVED** [review]
PTAL at couple tiny comments and feel free to merge after

**Comment by h5law** [comment]
**NOTE** @Olshansk, @dylanlott is currently working on savepoints/rollbacks that touches the same code and this PR will probably only be used as a reference as he is changing the logic their. The new wrapper functions fix the bug with `Delete()` but need to be integrated with savepoints. Will wait on merging this until @dylanlott gives an update

**Comment by Olshansk** [comment]
This PR is already complete and ready to be merged in, while savepoints & rollbacks PR is 1+ weeks away. 

IMO we should merge this in and iterate on top of it. However, I will defer to @dylanlott to decide.

**Comment by dylanlott** [comment]
> This PR is already complete and ready to be merged in, while savepoints & rollbacks PR is 1+ weeks away. 
> 
> 
> 
> IMO we should merge this in and iterate on top of it. However, I will defer to @dylanlott to decide.

I agree with merging this and addressing the logic changes necessary for savepoints later. It shouldn't hold this up.

File: persistence/kvstore/kvstore_test.go
Comment (h5law) [reply persistence/kvstore/kvstore_test.go:35]: Have made them independent

**Pull request merged**
